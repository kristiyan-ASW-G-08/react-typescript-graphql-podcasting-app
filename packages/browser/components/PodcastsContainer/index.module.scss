@import '../../styles/variables.module.scss';
@import '../../styles/mediaQueries.module.scss';
@import '../../styles/mixins.module.scss';

.podcasts {
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-gap: 5rem;
  margin: 3rem 0 3rem 0;
  overflow: hidden;
  @include tablets {
    grid-template-columns: repeat(3, 1fr);
  }
  @include desktop {
    grid-template-columns: repeat(5, 1fr);
  }
}
