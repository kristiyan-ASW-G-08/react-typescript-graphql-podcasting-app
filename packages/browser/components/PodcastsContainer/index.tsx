import apolloClient from '../../apolloClient';
import { FC, useState } from 'react';
import getPodcastsQuery from '@/queries/getPodcasts';
import styles from './index.module.scss';
import PodcastMiniCard from '@/components/PodcastMiniCard';
import Podcast from '@/types/Podcast';

interface PodcastsContainerProps {
  podcasts: Podcast[];
}
const PodcastsContainer: FC<PodcastsContainerProps> = ({ podcasts }) => {
  return (
    <section className={styles.container}>
      <div data-testid="podcasts" className={styles.podcasts}>
        {podcasts.map(podcast => (
          <PodcastMiniCard
            data-testid={podcast._id}
            {...podcast}
            key={podcast._id}
          />
        ))}
      </div>
    </section>
  );
};

export default PodcastsContainer;
