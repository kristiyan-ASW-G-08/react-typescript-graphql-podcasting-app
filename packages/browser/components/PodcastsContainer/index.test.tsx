import React, { FC } from 'react';
import { render, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import PodcastsContainer from '.';

describe('PodcastsContainer', () => {
  it('render PodcastsContainer', () => {
    const podcast = {
      title: 'SomeTItle',
      website: 'https://somewebsite.testit',
      cover: 'somecover.svg',
      user: {
        username: 'SomeUser',
        _id: 'someId',
      },
    };
    expect.assertions(2);
    const podcasts = [
      { ...podcast, _id: '1' },
      { ...podcast, _id: '2' },
      { ...podcast, _id: '3 ' },
      { ...podcast, _id: '4' },
      { ...podcast, _id: '5' },
      { ...podcast, _id: '6' },
      { ...podcast, _id: '7' },
      { ...podcast, _id: '8' },
    ];
    const { getByTestId } = render(<PodcastsContainer podcasts={podcasts} />);
    const podcastsContainer = getByTestId('podcasts');
    expect(podcastsContainer).toBeInTheDocument();
    waitFor(() => {
      expect(podcastsContainer.children.length).toEqual(podcasts.length);
    });
  });
});
