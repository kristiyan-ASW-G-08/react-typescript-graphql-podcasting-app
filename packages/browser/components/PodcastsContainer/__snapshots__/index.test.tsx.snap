// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PodcastsContainer render PodcastsContainer 1`] = `
HTMLCollection [
  <div
    class="card"
  >
    <img
      alt=""
      src="undefined/somecover.svg"
    />
    <h4>
      SomeTItle
    </h4>
    <button>
      Learn More
    </button>
  </div>,
  <div
    class="card"
  >
    <img
      alt=""
      src="undefined/somecover.svg"
    />
    <h4>
      SomeTItle
    </h4>
    <button>
      Learn More
    </button>
  </div>,
  <div
    class="card"
  >
    <img
      alt=""
      src="undefined/somecover.svg"
    />
    <h4>
      SomeTItle
    </h4>
    <button>
      Learn More
    </button>
  </div>,
  <div
    class="card"
  >
    <img
      alt=""
      src="undefined/somecover.svg"
    />
    <h4>
      SomeTItle
    </h4>
    <button>
      Learn More
    </button>
  </div>,
  <div
    class="card"
  >
    <img
      alt=""
      src="undefined/somecover.svg"
    />
    <h4>
      SomeTItle
    </h4>
    <button>
      Learn More
    </button>
  </div>,
  <div
    class="card"
  >
    <img
      alt=""
      src="undefined/somecover.svg"
    />
    <h4>
      SomeTItle
    </h4>
    <button>
      Learn More
    </button>
  </div>,
  <div
    class="card"
  >
    <img
      alt=""
      src="undefined/somecover.svg"
    />
    <h4>
      SomeTItle
    </h4>
    <button>
      Learn More
    </button>
  </div>,
  <div
    class="card"
  >
    <img
      alt=""
      src="undefined/somecover.svg"
    />
    <h4>
      SomeTItle
    </h4>
    <button>
      Learn More
    </button>
  </div>,
]
`;
